### PR TITLE
Update interfaces.py documentation

### PR DESCRIPTION
**Pull request by tungen**
@lsc36 @yangarbiter  Please help review/comment/modify. Thanks.

### CODE REVIEW

**Comment by yangarbiter** [comment]
https://github.com/ntucllab/libact/blob/interface-documentation/libact/base/interfaces.py#L128

It doesn't seems the update function returns itself.

File: libact/base/interfaces.py [hunk 1]
```diff
@@ -22,41 +21,112 @@ def dataset(self):
...
+        entry_id : float
+            The index of the newly labeled sample.
+            
+        label : float
+            The label of the queried sample.
+        
+        Returns
+        -------
+        self : object
+            Returns self.
```
Comment (yangarbiter) [inline libact/base/interfaces.py]: It doesn't seems the update function always returns itself.

File: libact/base/interfaces.py [hunk 2]
```diff
@@ -22,41 +21,112 @@ def dataset(self):
...
 
     @abstractmethod
     def make_query(self):
+        """Return the index of the sample to be queried and labeled. Read-only.
+        
+        No modification to the internal states.
+        
+        Returns
+        -------
+        ask_id : float
```
Comment (yangarbiter) [inline libact/base/interfaces.py]: It should be integer?

File: libact/base/interfaces.py [hunk 3]
```diff
@@ -22,41 +21,112 @@ def dataset(self):
...
+        """Return the class labels for the input feature array. 
+    
+        Parameters
+        ----------
+        feautre : array, shape (n_features,)
+            The feature vector whose label is to queried.
+            
+        Returns
+        -------
+        label : float
```
Comment (yangarbiter) [inline libact/base/interfaces.py]: The type for label should depends on the dataset object given?

File: libact/base/interfaces.py [hunk 4]
```diff
@@ -22,41 +21,112 @@ def dataset(self):
...
+        """Train a model according to the given training dataset. 
+        
+        Parameters
+        ----------
+        dataset : Dataset object
+             The training dataset the model is to be trained on.
+
+        Returns
+        -------
+        self : object
```
Comment (yangarbiter) [inline libact/base/interfaces.py]: It doesn't seems the update function always returns itself.

**Comment by yangarbiter** [comment]
array-like structure in Python can be converted to arrays through the use of the array() function

it seems we have to check this if we want to write array-like in document

**Comment by lsc36** [comment]
@yangarbiter @tungen ready for merge?

File: libact/base/interfaces.py (hunk 4 above)
Comment (tungen) [inline libact/base/interfaces.py]: train calls fit in sklearn's model which returns self

**Comment by lsc36** [comment]
@yangarbiter @tungen please focus on this first since this is release blocking.

File: libact/base/interfaces.py [hunk 5]
```diff
@@ -22,41 +21,107 @@ def dataset(self):
...
+    Assign labels to the samples queried by QueryStrategies.
     """
-
     @abstractmethod
     def label(self, feature):
+        """Return the class labels for the input feature array. 
+    
+        Parameters
+        ----------
+        feautre : array-like, shape (n_features,)
```
Comment (lsc36) [inline libact/base/interfaces.py]: typo

File: libact/base/interfaces.py [hunk 6]
```diff
@@ -22,41 +21,107 @@ def dataset(self):
...
+        """
         pass
 
 
 class Model(metaclass=ABCMeta):
     """Classification Model
 
-    A Model is trained on a training dataset and produces a class-predicting
-    function for future features.
+    A Model retruns a class-predicting function for future samples after
```
Comment (lsc36) [inline libact/base/interfaces.py]: typo

**Pull request merged**